server/src/main/java/io/druid/server/QueryResource.java

```diff
@@ -236,6 +243,7 @@ public Object accumulate(Object accumulated, Object in)
 
       try {
         final Query theQuery = query;
+        final QueryToolChest theToolChest = toolChest;
```

Comment:
This is here because the query and toolChest are passed to anonymous classes at some point, and so the references need to be final. But the original references aren't final due to how the exception handling is done.

I didn't really want to mess with any of that in this PR.